types/transaction.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcjson"
+)
+
+// Transaction represents a Bitcoin transaction.
+//
+// Wraps btcjson.TxRawResult.
+type Transaction struct {
+	data  btcjson.TxRawResult
+	block *Block
+}
+
+// NewTransaction creates a new transaction from raw json result.
+func NewTransaction(tx btcjson.TxRawResult) *Transaction {
+	return &Transaction{
+		tx,
+		nil,
+	}
+}
+
+// NewTransactionWithBlock creates a new transaction from raw json result and its block.
+func NewTransactionWithBlock(tx btcjson.TxRawResult, block *Block) (*Transaction, error) {
+	return NewTransaction(tx).WithBlock(block)
+}
+
+// WithBlock stores a reference to the block that contains the transaction.
+func (tx *Transaction) WithBlock(block *Block) (*Transaction, error) {
+	if block == nil {
+		return tx, nil
+	}
```

Comment:
Should this maybe return an error? I'm just thinking in the case where someone thinks they're block is not nil (but it is), and they pass it into this function, current functionality might lead them to believe they set the block just fine when in fact they didn't.

Reply:
Yeah, you're right